src/reason_pprint_ast.ml

```diff
@@ -3578,10 +3548,12 @@ class printer  ()= object(self:'self)
   method formatJSXComponent componentName args =
     let rec processArguments arguments processedAttrs children =
       match arguments with
-      | ("", {pexp_desc = Pexp_construct (_, None)}) :: tail ->
+      | ("children", {pexp_desc = Pexp_construct (_, None)}) :: tail ->
```

Comment:
It's not really locked. Your suggestion is implemented in #917.
<Foo>bar</Foo> desugaring to `Foo.createElement children::[bar] () [@JSX]` rather than `Foo.createElement [bar][@JSX]` isn't really more prescriptive. I'd say on the contrary, it allows children to be partially applied too rather than making the assumption that it can never be.